extensions/notebook/src/dialog/addFileDialog.ts

```diff
@@ -0,0 +1,112 @@
+	private _dialog: azdata.window.Dialog;
+	public view: azdata.ModelView;
+	private _formModel: azdata.FormContainer;
+	private _fileNameInputBox: azdata.InputBoxComponent;
+	private _titleInputBox: azdata.InputBoxComponent;
+	private _saveLocationInputBox: azdata.TextComponent;
+	private _prompter: IPrompter;
+
+	constructor(private _tocManager: IBookTocManager, private _bookItem: BookTreeItem, private _extension: FileExtension) {
+		this._prompter = new CodeAdapter();
+	}
+
+	public async validatePath(folderPath: string, fileBasename: string): Promise<boolean> {
+		const destinationUri = path.join(folderPath, fileBasename);
+		if (await pathExists(destinationUri)) {
+			const doOverwrite = await confirmMessageDialog(this._prompter, loc.confirmOverwrite);
+			if (doOverwrite) {
+				return true;
+			}
+			return false;
+		}
+		return await pathExists(folderPath);
+	}
+
+	public async createDialog(): Promise<void> {
+		const dialogTitle = this._extension === FileExtension.Notebook ? loc.newNotebook : loc.newMarkdown;
+		this._dialog = azdata.window.createModelViewDialog(dialogTitle);
+		this._dialog.registerContent(async view => {
+			this.view = view;
+			this._fileNameInputBox = this.view.modelBuilder.inputBox()
+				.withProperties({
+					enabled: true,
+					width: '400px'
+				}).component();
+
+			this._titleInputBox = this.view.modelBuilder.inputBox()
+				.withProperties({
+					enabled: true,
+					width: '400px'
+				}).component();
+
+			this._saveLocationInputBox = this.view.modelBuilder.inputBox()
+				.withProperties({
+					value: this._bookItem.contextValue === BookTreeItemType.Book ? this._bookItem.rootContentPath : path.dirname(this._bookItem.resourceUri.fsPath),
+					enabled: false,
+					width: '400px'
+				}).component();
+
+			this._formModel = this.view.modelBuilder.formContainer()
+				.withFormItems([{
+					components: [
+						{
+							title: loc.title,
+							required: true,
+							component: this._titleInputBox
+						},
+						{
+							component: this._fileNameInputBox,
+							title: loc.fileName,
+							required: true
+						},
+						{
+							component: this._saveLocationInputBox,
+							title: loc.saveLocation,
+							required: false
+						}
+					],
+					title: ''
+				}]).component();
+			await this.view.initializeModel(this._formModel);
+		});
+		this._dialog.okButton.label = loc.add;
+		this._dialog.registerCloseValidator(async () => await this.createFile());
+		azdata.window.openDialog(this._dialog);
+	}
+
+	private async createFile(): Promise<boolean> {
+		try {
+			const dirPath = this._bookItem.contextValue === BookTreeItemType.Book ? this._bookItem.rootContentPath : path.dirname(this._bookItem.resourceUri.fsPath);
+			const filePath = path.join(dirPath, this._fileNameInputBox.value).concat(this._extension);
+			const isValid = await this.validatePath(dirPath, this._fileNameInputBox.value.concat(this._extension));
+			if (!isValid) {
+				throw (new Error(loc.msgSaveFolderError));
```

Comment:
This error seems too generic - there's a couple different things that could be wrong if validate returns false (such as user declining to overwrite). I'd just suggest having validatePath throw an error with the appropriate message as needed.